.github/workflows/ci_plugins.yaml

```diff
@@ -0,0 +1,104 @@
+          declare -A plugin_set
+          for file in "${changed_files[@]}"; do
+            if [[ "$file" =~ ^plugins/ && ! "$file" =~ ^plugins/template/ ]]; then
+              plugin_name=$(echo "$file" | cut -d '/' -f 2)
+              plugin_set[$plugin_name]=1
+            fi
+          done
+          plugins=("${!plugin_set[@]}")
+          matrix_json=$(printf '%s\n' "${plugins[@]}" | jq -R -s -c '{plugin: split("\n")[:-1]}')
+          echo "matrix_json=$matrix_json"
+          echo "Changed plugins: ${plugins[*]}"
+          echo "matrix=$matrix_json" >> $GITHUB_OUTPUT
+          echo "matrix_json: $matrix_json"
+
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  test_plugin:
+    needs: prepare_matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.prepare_matrix.outputs.matrix)}}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Cache Pip Packages
+        id: setup-python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+          cache: 'pip' # caching pip dependencies
+
+      - name: Cache Python Installation
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.pythonLocation }} # Cache the whole python installation dir.
+          key: ${{ matrix.plugin }}_${{ hashFiles('pyproject.toml', '*/pyproject.toml') }}
+
+
+      - name: Install SuperDuperDB Project
+        run: |
+          # Install core and testsuite dependencies on the cached python environment.
+          python -m pip install '.[test]'
+
+      - name: Install DevKit (docs, testing, etc)
+        run: |
+          make install_devkit
+
+      - name: Lint and type-check
+        run: |
+          make lint-and-type-check DIRECTORIES="plugins/${{ matrix.plugin }}"
+
+      - name: Install Plugin
+        run: |
+          python -m pip install 'plugins/${{ matrix.plugin }}[test]'
+
+      - name: Optionally run custom CI script
+        run: |
+          if [ -f "plugins/${{ matrix.plugin }}/.ci_extend.sh" ]; then
+            echo "Running custom CI script..."
+            bash ./plugins/${{ matrix.plugin }}/.ci_extend.sh
+          else
+            echo "No custom CI script found, skipping..."
+          fi
+   
+      - name: Plugin Testing
+        run: |
+          export PYTHONPATH=./
+          pytest --cov=superduper --cov-report=xml plugins/${{ matrix.plugin }}/plugin_test
+
+      - name: Optionally run the base testing
+        run: |
+          SUPERDUPER_CONFIG="plugins/${{ matrix.plugin }}/plugin_test/config.yaml"
+          if [ -f "$SUPERDUPER_CONFIG" ]; then
+            echo "Running the base testing..."
+            make unit_testing SUPERDUPER_CONFIG=$SUPERDUPER_CONFIG
+            make usecase_testing SUPERDUPER_CONFIG=$SUPERDUPER_CONFIG
+          else
+            echo "No config file found, skipping..."
+          fi
```

Comment:
The plugin located under the `plugin_test` folder, which includes a `config.yaml` file, not only runs its own tests but also uses this configuration file to run test cases for superduper, including both unittest and usecase tests.